### PR TITLE
dep: Deprecate characteristic functions containing camel case

### PR DESCRIPTION
**Pull request by cadenmyers13**
closes #185 

This PR:

1) Changes CF functions from camelCase to snake_case
1) Deprecates the `*CF2` functions as they were just wrappers
1) Adds tests for characteristic functions

I also snuck `constrainAsSpaceGroup` deprecation onto this PR since that is a frequently used function

### CODE REVIEW

File: tests/test_characteristicfunctions.py [hunk 1]
```diff
@@ -188,5 +191,314 @@ def testCylinder(sas_available):
...
+    )
+
+
+@pytest.mark.parametrize(
+    "input_sheet_thickness",
+    [
+        # C1: zero thickness.
+        # Expected: the characteristic function is zero, regardless of r.
+        0.0,
+        # C2: negative thickness.
```
Comment (sbillinge) [inline tests/test_characteristicfunctions.py]: Is this the behavior we want?  I wonder if this will make  the regression unstable in edge cases, because there are an infinite number of values for the thickness on the negative axis that will return the same result.  I think we may want to raise an exception, or maybe better, have this behavior so the fit doesn't keep crashing but generate a warning that there is a negative thickness or something?   If we do that, we don't want it spitting out a million of these warnings on every loop though.  Let's think a bit about this.  Also, similarly for some of the other edge cases that look like this (physically unrealistic parameters)

**Review by sbillinge — COMMENTED** [review]
please see my comment.  I love these tests, they are so clear to navigate!  but let's think a bit more carefully.

File: tests/test_characteristicfunctions.py (hunk 1 above)
Comment (cadenmyers13) [reply tests/test_characteristicfunctions.py]: @sbillinge After some discussion with Claude, this is what it said,

"
The instability isn't really "infinitely many thicknesses give the same answer." It's that the Jacobian column for that parameter is exactly zero on the whole negative axis. leastsq computes ∂residual/∂thickness by finite differences; on the plateau it gets 0, the normal-equations matrix goes singular, and the parameter is frozen wherever it landed while everything else refines around it. So it's an absorbing trap: once the fit steps negative it can never step back, and the covariance matrix for that parameter is meaningless.

This matters because LM routinely probes outside the physical domain — both in the finite-difference step and on overshooting trial steps. It usually needs to come back.
"

TLDR; when the CF param goes negative/non-physical, it causes the parameter to remain unchanged as the refinement continues around it. 

As a solution, it recommends to warn the user to use the `FitRecipe.get_soft_bounds` to restrict the parameter from going negative rather than crashing the whole refinement because it stepped in the negative direction and the output should go to zeros. See my latest push. I added more tests for this
